blockchain/block.go

```diff
@@ -28,6 +27,9 @@ type BlockHeader struct {
 	Time uint32
 	// Nonce starts at 0 and increments by 1 for every hash when mining
 	Nonce uint64
+	// ExtraData is an extra field that can be filled with arbitrary data to
+	// be stored in the block
+	ExtraData []byte
```

Comment:
I know there is some analogue to this in bitcoin. Interested, why are we adding it to our chain?